src/components/Reports/ReportEdit.tsx

```diff
@@ -122,6 +123,22 @@ export default function ReportEdit(): JSX.Element {
     buttonType: 'passive',
   });
 
+  /**
+   * Report update functions
+   */
+  const updateReport = (field: string, value: any) => {
+    if (report) {
+      setReport(
+        produce((draft) => {
```

Comment:
Brought in this function from the `immer` library. It automatically handles creating a deep copy, keeping track of updates, and returning it to update state.

Reply:
this is cool..
in case it helps.. we have a similar pattern in accession edit.. there's a `useForm` util that is used across the board.

Reply:
Oh, i'll go take a look at that. Thanks

Reply:
Ah, okay. `useForm` looks pretty nice, but I'm not sure how well it would play with an object like a report that is nested 3 or 4 levels deep. For instance, if I have an object,
```javascript
obj0 = {
    obj1: {
        key: value
    }
}
```
Then `useForm`, which employs the spread operator, would only make a copy of the `obj1` reference, but `key: value` would not be copied. I think this could be a problem because updating the "copy" would modify the original react state at that level, which is a no-no. If I'm understanding correctly. I used `produce` because it deep-copies the object through its entire hierarchy.

Reply:
>We could theoretically use produce inside useForm (for the deep copy).. or leverage lodash's deepClone..

Oh cool, yeah we could enhance `useForm` even more

Reply:
All good, we can keep things as-is and refactor later if we want to.
Btw, in this example.. is draft the root level object and field a hierarchical property or a leaf property?
Ex.
draft = { 
  abc: {
      def: 123
  }
}

if we want to change def, is the field = abc.def? or just def?

Reply:
nvm, i see it is leaf level from usage 👍

Reply:
Yes. I will need to make some deeper modifications, so `produce` will show its worth more in future updates